runtime/block_sync_default.cu

```diff
@@ -25,15 +25,21 @@ namespace block_sync {
 __forceinline__ __device__ void init() {}
 
 // Thread-block synchronization
+// barrier 0 is reserved for the default block synchronization
```

Comment:
The comment and code change reads a bit strange to me here.

Looks like it's a change in behavior, we are now using `barrier_id == 1` as the default. Isn't that considered `default block synchronization`?
Are we actively passing argument for `barrier_id` at this point? asking since I don't see anything changed in codegen yet.

Reply:
Revised for clarity:
```
barrier 0 is reserved for default block synchronization, e.g. ` __syncthreads()`.
bar.sync should use other barriers to avoid conflicts with ` __syncthreads()`.
```
Not passing the argument yet, but will be used when we have multiple compute warp groups.

Reply:
Why changing the default? If multiple syncs are issued simultaneously, IDs should be specified explicitly anyway.

Reply:
> Why changing the default? If multiple syncs are issued simultaneously, IDs should be specified explicitly anyway.

barrier 0 is reserved for default block synchronization, e.g. ` __syncthreads()`. I think it is a good practice to avoid using system reserved barrier. For example in [cutlass](https://github.com/NVIDIA/cutlass/blob/ad7b2f5e84fcfa124cb02b91d5bd26d238c0459e/include/cutlass/arch/barrier.h#L157),  barrier `0` is also not used.

If we really want to start from `0`, we need to further revise our code to keep a record of all the syncs including `__syncthreads()`, doesn't seem a good ROI.

Reply:
It is not reserved. It is just used by syncthreads. To avoid conflicts, it's a bad idea to not specify the barrier ID. If conflicts aren't an issue, any barrier should be fine.